### PR TITLE
Initial tun term impl

### PR DESCRIPTION
**Pull request by DanG100**
working on test in new PR

### CODE REVIEW

File: dataplane/saiserver/switch.go [hunk 1]
```diff
@@ -93,6 +93,7 @@ const (
 	MyMacTable            = "my-mac-table"
 	hostifToPortTable     = "cpu-input"
 	portToHostifTable     = "cpu-output"
+	turnTermTable         = "turn-term"
```
Comment (guoshiuan) [inline dataplane/saiserver/switch.go:96]: tun-term

File: dataplane/saiserver/routing_test.go [hunk 2]
```diff
@@ -570,6 +570,7 @@ func TestCreateNextHop(t *testing.T) {
 		t.Run(tt.desc, func(t *testing.T) {
 			dplane := &fakeSwitchDataplane{}
 			c, mgr, stopFn := newTestNextHop(t, dplane)
+			mgr.StoreAttributes(10, &saipb.TunnelAttribute{Type: saipb.TunnelType_TUNNEL_TYPE_IPINIP.Enum()})
```
Comment (guoshiuan) [inline dataplane/saiserver/routing_test.go:573]: From the test cases, 10 is RIF ID and also tunnel ID. So we only store it for tunnel?

File: dataplane/saiserver/tunnel.go [hunk 3]
```diff
@@ -97,3 +97,127 @@ func (t *tunnel) CreateTunnel(ctx context.Context, req *saipb.CreateTunnelReques
...
+	ipV6ExactMask = []byte{0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF}
+	ipV6AnyMask   = make([]byte, 16)
+)
+
+func (t *tunnel) CreateTunnelTermTableEntry(ctx context.Context, req *saipb.CreateTunnelTermTableEntryRequest) (*saipb.CreateTunnelTermTableEntryResponse, error) {
+	id := t.mgr.NextID()
+
+	fields := []*fwdpb.PacketFieldMaskedBytes{}
+
+	// It is valid for some request fields to be omited, so initialize slices to the correct length for the given IP protocol.
```
Comment (guoshiuan) [inline dataplane/saiserver/tunnel.go:113]: omitted

File: dataplane/saiserver/switch.go (hunk 1 above)
Comment (DanG100) [reply dataplane/saiserver/switch.go]: done

File: dataplane/saiserver/routing_test.go (hunk 2 above)
Comment (DanG100) [reply dataplane/saiserver/routing_test.go]: changed to unique id

File: dataplane/saiserver/tunnel.go (hunk 3 above)
Comment (DanG100) [reply dataplane/saiserver/tunnel.go]: done

File: dataplane/saiserver/ports.go [hunk 4]
```diff
@@ -79,6 +79,7 @@ func getForwardingPipeline() []*fwdpb.ActionDesc {
 		fwdconfig.Action(fwdconfig.LookupAction(IngressVRFTable)).Build(),                               // Match interface to VRF.
 		fwdconfig.Action(fwdconfig.LookupAction(PreIngressActionTable)).Build(),                         // Run pre-ingress actions.
 		fwdconfig.Action(fwdconfig.DecapAction(fwdpb.PacketHeaderId_PACKET_HEADER_ID_ETHERNET)).Build(), // Decap L2 header.
+		fwdconfig.Action(fwdconfig.LookupAction(turnTermTable)).Build(),                                 // Decap the packet if we have a tunnel.
```
Comment (guoshiuan) [inline dataplane/saiserver/ports.go:82]: s/turnTermTable/tunTermTable/g

Comment (DanG100) [reply dataplane/saiserver/ports.go]: fixed

**Review by guoshiuan — APPROVED** [review]

**Pull request merged**